lib/results_validators/advancement_conditions_validator.rb

```diff
@@ -94,11 +94,13 @@ def run_validation(validator_data)
 
                 # Check that no one proceeded if they shouldn't have
                 if condition.instance_of? AdvancementConditions::AttemptResultCondition
-                  current_persons = results_by_round_type_id[round_type_id].map(&:wca_id)
+                  current_persons = results_by_round_type_id[round_type_id].map(&:person_id)
                   people_over_condition = previous_results.filter do |r|
                     sort_by_column = r.format.sort_by == "single" ? :best : :average
-                    current_persons.include?(r.wca_id) && r.send(sort_by_column) > condition.attempt_result
-                  end.map(&:wca_id)
+                    current_persons.include?(r.person_id) && r.send(sort_by_column) > condition.attempt_result
+                  end.map do |competitor|
```

Comment:
`person` is the intuitive name, but using `competitor` instead to avoid mislabelling, as the object can be (as I understand) either `person` (in case of Result) or `inbox_person` (in case of InboxResult)